### PR TITLE
Add account use cases

### PR DESCRIPTION
**Pull request by AnnaGolosova**
### *What is the problem we're trying to solve*

Adding logic to save new account and update existing account

### *What changes have we introduced*
Account balance updates with account updated event.
#### _Checklist_
- [x] Added queryable factories
- [x] Added account updated event in ElasticSearch
- [x] Added account created event
- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

File: HousingSearchListener/V1/Gateway/Interfaces/IEsGateway.cs [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
 namespace HousingSearchListener.V1.Gateway.Interfaces
 {
     public interface IEsGateway
     {
+        Task<IndexResponse> IndexAccount(QueryableAccount esAccount);
+
         Task<IndexResponse> IndexPerson(QueryablePerson esPerson);
 
-        Task<IndexResponse> IndexTenure(QueryableTenure esTenure);
+        Task<IndexResponse> IndexTenure(Hackney.Shared.HousingSearch.Gateways.Models.Tenures.QueryableTenure esTenure);
```
Comment (LBHMGeorgieva) [inline HousingSearchListener/V1/Gateway/Interfaces/IEsGateway.cs:17]: Hi Hanna, why are we using the full namespace path in so many places now like in this example?

File: HousingSearchListener/V1/Factories/QueryableFactories/TenuresFactory.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            {
+                Id = tenure.Id,
+                EndOfTenureDate = tenure.EndOfTenureDate,
+                PaymentReference = tenure.PaymentReference,
+                StartOfTenureDate = tenure.StartOfTenureDate,
+                Type = tenure.TenureType.Description
+            };
+        }
+
+        public QueryableTenure CreateQueryableTenure(TenureInformation tenure)
```
Comment (LBHMGeorgieva) [inline HousingSearchListener/V1/Factories/QueryableFactories/TenuresFactory.cs:28]: Hi Hanna, I believe this method exists also in the ESEntityFactory - could you confirm if it is actually needed in both places?

File: HousingSearchListener/V1/Factories/QueryableFactories/TransactionsFactory.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                    ResolutionDate = transaction.SuspenseResolutionInfo.ResolutionDate
+                } : null,
+                TargetId = transaction.TargetId,
+                TargetType = transaction.TargetType,
+                TransactionAmount = transaction.TransactionAmount,
+                TransactionDate = transaction.TransactionDate,
+                TransactionSource = transaction.TransactionSource,
+                TransactionType = transaction.TransactionType,
+                SortCode = transaction.SortCode,
+                CreatedAt = transaction.CreatedAt,
```
Comment (LBHMGeorgieva) [inline HousingSearchListener/V1/Factories/QueryableFactories/TransactionsFactory.cs:47]: Hi Hanna, could you please confirm if all of those fields are needed for the Transactions entity - will they all be used for searching transactions / displaying transactions search response?

File: HousingSearchListener/V1/Infrastructure/QueryResponseExtension.cs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+                    ConsolidatedCharges = consolidatedChargesList,
+                    Tenure = tenure,
+                    TargetType = Enum.Parse<TargetType>(item["target_type"].S),
+                    TargetId = Guid.Parse(item["target_id"].S),
+                    AccountType = Enum.Parse<AccountType>(item["account_type"].S),
+                    RentGroupType = Enum.Parse<RentGroupType>(item["rent_group_type"].S),
+                    AgreementType = item.ContainsKey("agreement_type") ? item["agreement_type"].S : null,
+                    CreatedBy = item.ContainsKey("created_by") ? item["created_by"].S : null,
+                    LastUpdatedBy = item.ContainsKey("last_updated_by") ? item["last_updated_by"].S : null,
+                    CreatedAt = DateTime.Parse(item["created_at"].S),
```
Comment (LBHMGeorgieva) [inline HousingSearchListener/V1/Infrastructure/QueryResponseExtension.cs:76]: Hi Hanna, could you please confirm if all of those fields are needed for the accouns entity - will they all be used for searching accounts / displaying accounts search response?

**Review by LBHMGeorgieva — COMMENTED** [review]
Hi Hanna - I left a few comments with questions.

The PR currently changes quite a few files and it makes it more difficult to follow and review. Can you please separate the PR - one for updating account use case and one for creating?

Also, what was the reason behind creating multiple factory classes (e.g. TenuresFactory) instead of the ESEntityFactory?

File: HousingSearchListener/V1/Gateway/Interfaces/IEsGateway.cs (hunk 1 above)
Comment (AnnaGolosova) [reply HousingSearchListener/V1/Gateway/Interfaces/IEsGateway.cs:17]: Hi! We have two different QueryableTenure models. One as submodel for Accounts, one for Tenures

File: HousingSearchListener/V1/Factories/QueryableFactories/TransactionsFactory.cs (hunk 3 above)
Comment (AnnaGolosova) [reply HousingSearchListener/V1/Factories/QueryableFactories/TransactionsFactory.cs:47]: Hi! CreatedAt will be used to filter transactions by date. Some of fields willl be used on FE to display on the screen. And also we're using HS API to ge transactions from some listeners/nighty jobs

**Comment by AnnaGolosova** [comment]
@LBHMGeorgieva Hi! About your question:
`What was the reason behind creating multiple factory classes (e.g. TenuresFactory) instead of the ESEntityFactory?`

Now we have a lot of entities in out listener (Transactions, Accounts, Tenures, Assets etc.).
And for each entity we need some mapping. And that's why I splitted a big ESEntityFactory class to several small classes. Not each factory class contains logic related to only one entity. This way Open-Close SOLID principle willn't be violated

About PR splitting - no problem, I will split logic

**Pull request closed without merging**